frontend/src/components/pages/your_clubs/CreateNewClub.tsx

```diff
@@ -0,0 +1,242 @@
+    const contentWarningsArray = (event.target.value).split('\n');
+    setClub({...club, contentWarnings: contentWarningsArray});
+  }
+
+  const handleBookTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, title: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleBookAuthorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, author: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleBookIsbnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, isbn: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleCloseCreate = (successfulCreation: boolean) => {
+    setClub(undefined);
+    setBook(undefined);
+    setContentWarningsDisplay(undefined);
+    setMissingField(false);
+    setSubmitSuccess(false);
+    props.closeWindow(successfulCreation);
+  }
+
+  const handleClubSubmission = async () => {
+    if (!(club?.name && club?.description && book?.title && book?.author)) {
+      setMissingField(true);
+      setSubmitSuccess(false);
+    } else {
+      setMissingField(false);
+      const success = yourClubsHandlerService.createClub(club);
+      if (success) {
+        setSubmitSuccess(true);
+      }
+    }
+  }
+
+  return (
+    <div>
+      <Dialog open={props.isCreatingNewClub} maxWidth="lg">
+        <DialogTitle>Create New Club</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            {submitSuccess
+              ? <b>Submission successful.</b>
+              : "By filling out the below fields and clicking 'Submit', a club \
+                 of which you have administrative rights over will be created." }
+          </DialogContentText>
+          <form>
+            <TextField
+              error = {
+                missingField
+                  ? (club ? (club.name ? false : true) : true)
+                  : false
+              }
+              fullWidth
+              helperText = {missingField
+                ? club
+                  ? (club.name
+                    ? "Name of the club to be displayed publically to \
+                       all subsequent users who view it."
+                    : "Club Name is a required field.")
+                  : "Club Name is a required field."
+                : "Name of the club to be displayed publically to \
+                   all subsequent users who view it."}
+
+              id="clubName"
+              InputLabelProps={{
+                shrink: true,
+              }}
+              label="Club Name"
+              margin="normal"
+              onChange={handleClubNameChange}
+              placeholder="Club Name"
+              required
+              value={club ? club.name : ""}
+              variant="outlined"
+            />
+            <TextField
+              error = {missingField && !club?.description ? true : false}
```

Comment:
nit: `missingField && !club?.description` is already a boolean, no need to make it a ternary :)